src/game.cpp

```diff
@@ -3000,33 +3000,33 @@ bool game::handle_action()
 
 #define SCENT_RADIUS 40
 
-int &game::scent(int x, int y)
-{
-    if (x < (SEEX * MAPSIZE / 2) - SCENT_RADIUS || x >= (SEEX * MAPSIZE / 2) + SCENT_RADIUS ||
-        y < (SEEY * MAPSIZE / 2) - SCENT_RADIUS || y >= (SEEY * MAPSIZE / 2) + SCENT_RADIUS) {
-        nulscent = 0;
-        return nulscent; // Out-of-bounds - null scent
-    }
-    return grscent[x][y];
+// TODO: Unify this and scent area used in update_scent to fix "scent pocket" bug
+bool outside_scent_radius( const tripoint &p ) {
+    return p.x < (SEEX * MAPSIZE / 2) - SCENT_RADIUS || p.x >= (SEEX * MAPSIZE / 2) + SCENT_RADIUS ||
+           p.y < (SEEY * MAPSIZE / 2) - SCENT_RADIUS || p.y >= (SEEY * MAPSIZE / 2) + SCENT_RADIUS;
 }
 
 int &game::scent( const tripoint &p ) // A wrapper for now
```

Comment:
The comment refers to the fact that area in `outside_scent_radius` is different from the one in `update_scent`.

`update_scent` has an area like `u.pos() +/- SCENT_RADIUS`, while `outside_scent_radius` has a hardcoded dependence on the center of the map. This discrepancy results in "pockets" of scent that do not decay.
I didn't fix this one here, because centering the scent map on the center of the bubble could result in jarring changes, while centering around the player would slow down scent processing.